libs/application/templates/new-primary-school/src/fields/Review/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { useMutation } from '@apollo/client'
+import { SUBMIT_APPLICATION } from '@island.is/application/graphql'
+import {
+  Application,
+  DefaultEvents,
+  Field,
+  RecordObject,
+} from '@island.is/application/types'
+import { handleServerError } from '@island.is/application/ui-components'
+import { Box, Button, Text } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import get from 'lodash/get'
+import has from 'lodash/has'
+import { FC } from 'react'
+import { ReasonForApplicationOptions, States } from '../../lib/constants'
+import { newPrimarySchoolMessages } from '../../lib/messages'
+import { getApplicationAnswers } from '../../lib/newPrimarySchoolUtils'
+
+import { AllergiesAndIntolerances } from './review-groups/AllergiesAndIntolerances'
+import { Child } from './review-groups/Child'
+import { Languages } from './review-groups/Languages'
+import { Parents } from './review-groups/Parents'
+import { Photography } from './review-groups/Photography'
+import { ReasonForApplication } from './review-groups/ReasonForApplication'
+import { Relatives } from './review-groups/Relatives'
+import { Siblings } from './review-groups/Siblings'
+import { Support } from './review-groups/Support'
+import { School } from './review-groups/School'
+
+interface ReviewScreenProps {
+  application: Application
+  field: Field & { props?: { editable?: boolean } }
+  goToScreen?: (id: string) => void
+  refetch?: () => void
+  errors?: RecordObject
+  editable?: boolean
+}
+
+export const Review: FC<ReviewScreenProps> = ({
+  application,
+  field,
+  goToScreen,
+  refetch,
+  errors,
+}) => {
+  const { formatMessage } = useLocale()
+  const editable = field.props?.editable ?? false
+  const hasError = (id: string) => get(errors, id) as string
+
+  const { reasonForApplication } = getApplicationAnswers(application.answers)
+
+  const groupHasNoErrors = (ids: string[]) =>
+    ids.every((id) => !has(errors, id))
+
+  const childProps = {
+    application,
+    editable,
+    groupHasNoErrors,
+    hasError,
+    goToScreen,
+  }
+
+  const { state } = application
+
+  const [submitApplication, { loading: loadingSubmit }] = useMutation(
+    SUBMIT_APPLICATION,
+    {
+      onError: (e) => handleServerError(e, formatMessage),
+    },
+  )
```

Comment:
Is it possible to use `buildSubmitField` or does the custom submit logic don't fit into it?

Reply:
We need this for now. But we will try to improve this in the future.